engine/gamesys/src/gamesys/components/comp_model.cpp

```diff
@@ -1224,15 +1231,36 @@ namespace dmGameSystem
     }
     #endif
 
+    static inline bool CanUseDefaultVertexDeclaration(dmRender::HMaterial material)
+    {
+        const dmGraphics::VertexAttribute* attributes = 0;
+        uint32_t attribute_count = 0;
+        dmRender::GetMaterialProgramAttributes(material, &attributes, &attribute_count);
+        for (int i = 0; i < attribute_count; ++i)
+        {
+            const dmGraphics::VertexAttribute& attr = attributes[i];
+            if (attr.m_DataType != dmGraphics::VertexAttribute::TYPE_FLOAT)
+            {
+                return false;
+            }
+            if (!IsDefaultStream(attr.m_NameHash, attr.m_SemanticType, dmGraphics::VERTEX_STEP_FUNCTION_VERTEX))
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
     static void PrepareWorldSpaceBatchBuffers(ModelWorld* world, uint32_t batch_index, dmRender::HMaterial material,
         dmGraphics::HVertexDeclaration* vx_decl_in_out, uint32_t* vertex_stride_out,
         dmGraphics::VertexAttributeInfos* material_infos_vertex, uint32_t vertex_count, uint8_t** vb_begin)
     {
         *vb_begin = 0;
         memset(material_infos_vertex, 0, sizeof(dmGraphics::VertexAttributeInfos));
+
         dmGraphics::HVertexDeclaration vx_decl = *vx_decl_in_out;
 
-        if (!HasCustomVertexAttributes(material))
+        if (CanUseDefaultVertexDeclaration(material))
```

Comment:
This slightly differs because we only want to check if the vertex declaration contains regular "vertex" spaced attributes and not instanced. This fixes the case where someone is using a "mtx_world" attribute but no instancing is enabled (or supported), and we can use the fast route.